dask/delayed.py

```diff
@@ -86,7 +86,7 @@ def to_task_dask(expr):
             return (type(expr), args), dsk
         else:
             return args, dsk
-    if isinstance(expr, dict):
+    if type(expr) is dict:
```

Comment:
Would it be better to check something that handles dict-like things? This will still fail for something like

```
class MyClass2(collections.UserDict):
    pass
```

which I think is the recommended base class for dict-like classes.